tests/p2p/test_server.py

```diff
@@ -60,13 +93,49 @@ def server():
     assert len(server.peer_pool.connected_nodes) == 1
     initiator_peer = server.peer_pool.connected_nodes[INITIATOR_REMOTE]
     assert isinstance(initiator_peer, ETHPeer)
-    assert initiator_peer.privkey == keys.PrivateKey(eip8_values['receiver_private_key'])
+    assert initiator_peer.privkey == RECEIVER_PRIVKEY
+    await server.stop()
+    await asyncio.sleep(1)
 
-    server._server.close()
-    await server._server.wait_closed()
 
+@pytest.mark.asyncio
+async def test_two_servers(event_loop):
+    # Start server.
+    server_1 = get_server(
+        RECEIVER_PRIVKEY,
+        SERVER_ADDRESS,
+    )
+    bootstrap_nodes = [
+        'enode://{}@{}:{}'.format(
+            RECEIVER_PUBKEY.to_hex()[2:],
+            SERVER_ADDRESS.ip,
+            SERVER_ADDRESS.tcp_port
+        )
+    ]
+    # Set server_1 as server_2's boostrape node.
+    server_2 = get_server(
+        INITIATOR_PRIVKEY,
+        INITIATOR_ADDRESS,
+        bootstrap_nodes=bootstrap_nodes,
+    )
+
+    await server_1.start()
+    await server_2.start()
+
+    # server_2 will try to connect to server_1.
+    asyncio.ensure_future(server_2.peer_pool.run())
 
-async def send_auth_msg_to_server(address, messages):
-    reader, writer = await asyncio.open_connection(address.ip, address.udp_port)
-    writer.write(messages)
-    await writer.drain()
+    await asyncio.sleep(0.2)
```

Comment:
Can we await for an event or something else that would indicate the peers have connected? Ideally we should have no sleep()s in tests as those make them very fragile

Reply:
@gsalgado In order to remove the sleep()s, I changed to test the connecting via calling `peer_pool._connect_to_nodes(nodes)` directly instead of setting bootstrap nodes.